cpp/ql/src/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -98,51 +101,32 @@ class IteratorCrementOperator extends Operator, DataFlowFunction {
  * A non-member `operator+` function for an iterator type.
  */
 class IteratorAddOperator extends Operator, TaintFunction {
+  int iteratorIndex;
+
   IteratorAddOperator() {
     this.hasName("operator+") and
-    (
-      this
-          .getACallToThisFunction()
-          .getArgument(0)
-          .getFullyConverted()
-          .getUnspecifiedType()
-          .(PointerType)
-          .getBaseType() instanceof Iterator or
-      this
-          .getACallToThisFunction()
-          .getArgument(0)
-          .getFullyConverted()
-          .getUnspecifiedType()
-          .(PointerType)
-          .getBaseType() instanceof Iterator
-    )
+    iteratorIndex = [0, 1] and
+    calledWithIteratorArgument(this, iteratorIndex)
   }
 
   override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
-    (
-      input.isParameter(0) or
-      input.isParameter(1)
-    ) and
+    input.isParameter(iteratorIndex) and
```

Comment:
For the reference case of `calledWithIteratorArgument`, this should be `isParameterDeref`. Same below. Perhaps that can be achieved by changing the `int` parameter of `calledWithIteratorArgument` to be a `FunctionInput`.